pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotRealtimeSegmentManager.java

```diff
@@ -138,15 +138,22 @@ private synchronized void assignRealtimeSegmentsToServerInstancesIfNecessary()
         for (String instanceId : instancesInResource) {
           InstanceZKMetadata instanceZKMetadata = _pinotHelixResourceManager.getInstanceZKMetadata(instanceId);
           String groupId = instanceZKMetadata.getGroupId(resource);
+          String partitionId = instanceZKMetadata.getPartition(resource);
+          AbstractTableConfig realtimeTableConfig =
+              ZKMetadataProvider.getRealtimeTableConfig(_pinotHelixResourceManager.getPropertyStore(), resource);
+          if (instancesInResource.size() == Integer.parseInt(realtimeTableConfig.getValidationConfig().getReplication())) {
```

Comment:
In this case, I would remove the check and have the partition ids change to the value in instance config. The partition id is not used at the moment by the routing table builder, so it's fine if it changes at this moment. We should change the values assigned to the partition id on a per instance basis when we assign them, but this can be done in another PR.